frontend/src/app/services/events.ts

```diff
@@ -49,19 +62,40 @@ export class EventHandler {
     stabilityCost: 0,
     vaccinationPerDay: 0,
   };
-  triggerStateHistory: Record<string, TriggerState[]> = {};
+  eventStateHistory: Record<string, EventState> = {};
 
-  evaluateDay(prevDate: string, currentDate: string, dayState: DayState) {
-    let prevState = this.triggerStateHistory[prevDate];
+  evaluateDay(prevDate: string, currentDate: string, dayState: DayState, eventMitigations: EventMitigation[]) {
+    let prevState = this.eventStateHistory[prevDate];
     if (!prevState) {
-      prevState = eventTriggers.map(et => ({trigger: et, timeout: 0}));
+      let autoid = 0;
+      const triggerStates: Record<string, TriggerState> = {};
+      eventTriggers.forEach(et => {
+        let id = `autoid-${autoid++}`;
```

Comment:
I do not understand why do you generate missing ids, I think you are not using them effectively anywhere.

Reply:
I just need some key to store the TriggerStates to the dictionary if the user doesn't spec one explicitly. The autogenerated ones are indeed not used anywhere - is there any better way to do this?

Reply:
Sry, I just do not see the reason you need the dict for. Would not array as before be enough?

Reply:
So that i can reference specific trigger states from the code. E.g. do not complain about 100 deaths if the game already warned me about 200 deaths (the testers found some pathological cases like that). Or more generally do not warn me about one thing too soon after another.